dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/impl/QuiesceHandler.java

```diff
@@ -10,25 +10,34 @@
 package io.openliberty.netty.internal.impl;
 
 import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
 
 import com.ibm.websphere.ras.Tr;
 import com.ibm.websphere.ras.TraceComponent;
 
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelFutureListener;
 import io.netty.channel.SimpleUserEventChannelHandler;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.handler.codec.http.websocketx.CloseWebSocketFrame;
+import io.netty.util.AttributeKey;
+import io.netty.channel.ChannelPromise;
+
+import io.openliberty.netty.internal.impl.QuiesceState;
 
 /**
  * Channel handler which is added to the pipeline to terminates new connections once the 
  * quiesce period is started.
- *
  */
-public class QuiesceHandler extends SimpleUserEventChannelHandler<QuiesceHandler.QuiesceEvent>{
-
-	static class QuiesceEvent{
+public class QuiesceHandler extends ChannelDuplexHandler{
 
-	}
+	
 	
 	public static final QuiesceEvent QUIESCE_EVENT = new QuiesceEvent();
+	public static final AttributeKey<Boolean> WEBSOCKET_ATTR_KEY = AttributeKey.valueOf("websocket");
```

Comment:
If we can move this wsoc logic maybe to the netty chain quiesce I think that'd be best to avoid cross contamination here